components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -295,6 +297,7 @@ RewardsServiceImpl::RewardsServiceImpl(Profile* profile)
       publisher_state_path_(profile_->GetPath().Append(kPublisher_state)),
       publisher_info_db_path_(profile->GetPath().Append(kPublisher_info_db)),
       publisher_list_path_(profile->GetPath().Append(kPublishers_list)),
+      ads_service_(brave_ads::AdsServiceFactory::GetForProfile(profile_)),
```

Comment:
should we only add ads_service when ads are enabled (check build flag)? @bridiver

Reply:
we actually can't have this here at all. AdsService depends on RewardsService. @ryanml we need to go back to having all of this in ads service. This breaks the dependency and `ads_service_` could be null during shutdown. I'm surprised you don't get a DCHECK error here

Reply:
@NejcZdovc is partially correct here. AdsServiceFactory is actually exposed even when ads service is disabled by build flag, but `ads_service_` will be null in that case